tests/test_skewing.py

```diff
@@ -0,0 +1,393 @@
+
+from devito import Grid, TimeFunction, Eq, Operator, configuration, switchconfig, Le, Lt, Ge, Gt  # noqa
+
+from devito.core.autotuning import options  # noqa
+from devito import (NODE, Eq, Inc, Constant, Function, TimeFunction, SparseTimeFunction,  # noqa
+                    Dimension, SubDimension, Grid, Operator, norm, grad, div, dimensions,
+                    switchconfig, configuration, centered, first_derivative, solve,
+                    transpose)
+from devito.ir import (Expression, Iteration, FindNodes,
+                       retrieve_iteration_tree, Call, Conditional)
+from devito.tools import as_tuple
+
+
+def get_blocksizes(op, opt, grid, blockshape, level=0):
+    blocksizes = {'%s0_blk%d_size' % (d, level): v
+                  for d, v in zip(grid.dimensions, blockshape)}
+    blocksizes = {k: v for k, v in blocksizes.items() if k in op._known_arguments}
+    # Sanity check
+    if grid.dim == 1 or len(blockshape) == 0:
+        assert len(blocksizes) == 0
+        return {}
+    try:
+        if opt[1].get('blockinner'):
+            assert len(blocksizes) >= 1
+            if grid.dim == len(blockshape):
+                assert len(blocksizes) == len(blockshape)
+            else:
+                assert len(blocksizes) <= len(blockshape)
+        return blocksizes
+    except AttributeError:
+        assert len(blocksizes) == 0
+        return {}
+
+
+def _new_operator2(shape, time_order, blockshape=None, opt=None):
+    blockshape = as_tuple(blockshape)
+    grid = Grid(shape=shape, dtype=np.int32)
+    infield = TimeFunction(name='infield', grid=grid, time_order=time_order)
+    infield.data[:] = np.arange(reduce(mul, shape), dtype=np.int32).reshape(shape)
+    outfield = TimeFunction(name='outfield', grid=grid, time_order=time_order)
+
+    stencil = Eq(outfield.forward.indexify(),
+                 outfield.indexify() + infield.indexify()*3.0)
+    op = Operator(stencil, opt=opt)
+
+    blocksizes = get_blocksizes(op, opt, grid, blockshape)
+    op(infield=infield, outfield=outfield, t=10, **blocksizes)
+
+    return outfield, op
+
+
+def _new_operator3(shape, blockshape0=None, blockshape1=None, opt=None):
+    blockshape0 = as_tuple(blockshape0)
+    blockshape1 = as_tuple(blockshape1)
+
+    grid = Grid(shape=shape, extent=shape, dtype=np.float64)
+
+    # Allocate the grid and set initial condition
+    # Note: This should be made simpler through the use of defaults
+    u = TimeFunction(name='u', grid=grid, time_order=1, space_order=(2, 2, 2))
+    u.data[0, :] = np.linspace(-1, 1, reduce(mul, shape)).reshape(shape)
+
+    # Derive the stencil according to devito conventions
+    op = Operator(Eq(u.forward, 0.5 * u.laplace + u), opt=opt)
+
+    blocksizes0 = get_blocksizes(op, opt, grid, blockshape0, 0)
+    blocksizes1 = get_blocksizes(op, opt, grid, blockshape1, 1)
+    op.apply(u=u, t=10, **blocksizes0, **blocksizes1)
+
+    return u.data[1, :], op
+
+
+class TestAutotuningWithSkewing(object):
```

Comment:
No, too much code replication. Maybe what you actually want to do is parametrizing tests in test_autotuning?

Reply:
and, if necessary, add a few ones there? autotuning is a very specific features, and anything concerning autotuning should go there I feel

Reply:
Sure, I agree, I was aware of that. I just added a lot of duplicate/redundant test code for this draft state.
Will fix accordingly DSE/DLE insprired ones

Reply:
OK, poke me once done